### PR TITLE
Add class give-disable which return false and add some css  and run g…

### LINKED ISSUE #2664: Remove unwanted hidden fields from Donation Process Page
## Issue Overview
If guess donation is Disable then there is an extra input element is add which has the same name as checkbox name

image
image

Video Link: https://screencast-o-matic.com/watch/cFVb6xoD7u

## Expected Behavior
Should not have the same name

## Current Behavior
Have the same name

## Steps to Reproduce (for bugs)
See the Video

## Todos
- [x] Tests
- [x] Documentation

## WordPress Environment
<details>
```
### WordPress Environment ###

Home URL: http://give18.local
Site URL: http://give18.local
WP Version: 4.9.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
Show on Front: posts
Table Prefix Length: 3
Table Prefix Status: Acceptable
Admin AJAX: Accessible
Registered Post Statuses: publish, future, draft, pending, private, trash, auto-draft, inherit, refunded, failed, revoked, cancelled, abandoned, processing, preapproval

### Server Environment ###

Hosting Provider: DBH: localhost, SRV: give18.local
TLS Connection: Connection uses TLS 1.2
TLS Connection: Probably Okay
Server Info: nginx/1.10.1
PHP Version: 7.0.3
PHP Post Max Size: 1,000 MB
PHP Time Limit: 900
PHP Max Input Vars: 3000
PHP Max Upload Size: 1,000 MB
cURL Version: ❌ 7.38.0, OpenSSL/1.0.1t - We recommend a minimum cURL version of 7.40.
SUHOSIN Installed: –
MySQL Version: ❌ 5.5.55 - We recommend a minimum MySQL version of 5.6. See: WordPress Requirements
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
gzip: ✔
GD Graphics Library: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Give Configuration ###

Give Version: 2.0.0
Give Cache: Enabled
Database Updates: All DB Updates Completed.
Give Cache: Enabled
Give Cache: ✔New Donation✔Donation Receipt✔New Offline Donation✔Offline Donation Instructions✔New Donor Registration✔Donor Register❌Email access
Upgraded From: –
Test Mode: Enabled
Currency Code: GBP
Currency Position: Before
Decimal Separator: .
Thousands Separator: ,
Success Page: http://give18.local/donation-confirmation/
Failure Page: http://give18.local/donation-failed/
Donation History Page: http://give18.local/donation-history/
Give Forms Slug: /donations/
Enabled Payment Gateways: Test Donation, Offline Donation
Default Payment Gateway: Test Donation
PayPal IPN Verification: Enabled
PayPal IPN Notifications: N/A
Donor Email Access: Disabled

### Session Configuration ###

Give Use Sessions: Enabled
Session: Disabled

### Active Give Add-ons ###


### Other Active Plugins ###

Give - Development Plugin: by Raftaar1191 – 1.0.0

### Inactive Plugins ###

Duplicate Post: by Enrico Battocchi – 3.2.1
Give - 2Checkout Gateway: by WordImpress – 1.0.2
Give - Authorize.net Gateway: by WordImpress – 1.3.2
Give - AWeber: by WordImpress – 1.0.2
Give - Braintree Gateway: by WordImpress – 1.1.1
Give - CCAvenue Gateway: by WordImpress – 1.0
Give - ConvertKit: by WordImpress – 1.0.1
Give - CSV Toolbox: by WordImpress – 1.0
Give - Display Donors: by WordImpress, LLC – 1.0
Give - Dwolla Gateway: by WordImpress – 1.1.2
Give - Email Reports: by WordImpress – 1.0.2
Give - Fee Recovery: by WordImpress – 1.3.8
Give - Form Countdown: by WordImpress – 1.0
Give - Form Field Manager: by WordImpress – 1.2.4
Give - Gift Aid: by WordImpress – 1.0.0
Give - Google Analytics Donation Tracking: by WordImpress – 1.1.1
Give - MailChimp: by WordImpress – 1.3.7
Give - Manual Donations: by WordImpress – 1.2.2
Give - Paymill Gateway: by WordImpress – 1.1
Give - PDF Receipts: by WordImpress – 2.2.2
Give - Recurring Donations: by WordImpress – 1.5.4
Give - Stripe Gateway: by WordImpress – 1.5.2
Give - Tributes: by WordImpress – 1.3.3

### Theme ###

Name: Twenty Sixteen
Version: 1.4
Author URL: https://wordpress.org/
Child Theme: No – If you're modifying Give on a parent theme you didn't build personally, then we recommend using a child theme. See: How to Create a Child Theme

```
</details>

### PR DESCRIPTION
**Pull request by raftaar1191**
PR to Fix #2664

## Description
Remove unwanted hidden that was added because the serialized function of jquery does not allow the disable element value to pass

Video Link: https://screencast-o-matic.com/watch/cFVbXroDMZ

Notes: https://api.jquery.com/serialize/ does not serialize, The element with has disable attribute init

## How Has This Been Tested?
Manually Tested by adding new donation with enabling and disabling the guess donation option

## Screenshots (jpeg or gifs if applicable):
image

image

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: assets/scss/frontend/forms.scss [hunk 1]
```diff
@@ -282,6 +282,7 @@ form.give-form {
 	}
 	.give-disabled, [disabled] {
 		cursor: not-allowed;
+		opacity: 0.5;
```
Comment (ravinderk) [inline assets/scss/frontend/forms.scss]: @raftaar1191 why are you using opacity?

Comment (raftaar1191) [reply assets/scss/frontend/forms.scss]: @ravinderk  Look nice after adding that for UL purpose

Comment (raftaar1191) [reply assets/scss/frontend/forms.scss]: WIth Opacity:
image

Without Opacity:
image

Comment (ravinderk) [reply assets/scss/frontend/forms.scss]: @raftaar1191 `give-disabled` class we can use anywhere, it is not good to add styling to it. Just cursor property is fine. Another thing we or another developer can manage.

Let me if you are not getting my point.

Comment (raftaar1191) [reply assets/scss/frontend/forms.scss]: @ravinderk remove the opacity style from `give-disabled` 

Kindly please review this PR and merge

File: includes/forms/template.php [hunk 2]
```diff
@@ -1117,12 +1117,12 @@ function give_get_register_fields( $form_id ) {
 					if( ! give_is_setting_enabled( $is_guest_checkout ) ) {
 						echo Give()->tooltips->render(
 							array(
-								'tag_content' => '<input type="checkbox" name="give_create_account" id="give-create-account-" class="give-input" checked disabled /><input type="hidden" name="give_create_account" value="on" />',
+								'tag_content' => '<input type="checkbox" name="give_create_account" value="on" id="give-create-account-" class="give-input give-disabled" checked />',
```
Comment (mehul0810) [inline includes/forms/template.php]: @raftaar1191 Please disable this checkbox when `Guest Donations` are disabled as a donor can't must login to donate.

Comment (mehul0810) [reply includes/forms/template.php]: As discussed with @raftaar1191 over call, it is the issue due to caching. Then tested in FireFox, Chrome and Safari to double-check

@ravinderk This PR is working fine for me.

**Pull request merged**